elixir/apps/domain/priv/repo/migrations/20250717055156_create_processed_stripe_events.exs

```diff
@@ -0,0 +1,25 @@
+defmodule Domain.Repo.Migrations.CreateProcessedStripeEvents do
+  use Ecto.Migration
+
+  def up do
+    create table(:processed_stripe_events, primary_key: false) do
+      add(:stripe_event_id, :string, null: false, primary_key: true)
+      add(:event_type, :string, null: false)
+      add(:processed_at, :utc_datetime, null: false, default: fragment("NOW()"))
+      add(:stripe_customer_id, :string, null: true)
+      add(:event_created_at, :utc_datetime, null: false)
+      add(:livemode, :boolean, null: false, default: false)
+
+      timestamps(type: :utc_datetime_usec)
+    end
+
+    create(index(:processed_stripe_events, [:stripe_customer_id]))
+    create(index(:processed_stripe_events, [:event_type]))
+    create(index(:processed_stripe_events, [:event_created_at]))
+    create(index(:processed_stripe_events, [:stripe_customer_id, :event_type, :event_created_at]))
```

Comment:
Are all of these indexes necessary? Can we assume we'll always have a stripe_customer_id to scope queries to?

Reply:
Yeah that should work as long as we pass the stripe_customer_id always. That will cut the number of indexes in half here.